tensorboard/plugins/graph/tf_graph_common/util.ts

```diff
@@ -19,26 +19,44 @@ import * as _ from 'lodash';
 
 import {notifyActionEventFromPolymer} from '../../../components/tb_debug';
 import {
+  GraphDebugActionEventId,
   GraphDebugEventId,
+  GraphDebugTimingEventId,
+  GRAPH_DEBUG_ACTION_EVENT_CATEGORY,
   GRAPH_DEBUG_TIMING_EVENT_CATEGORY,
+  isDebugTimingEventId,
 } from '../../../components/tb_debug/types';
 import {NodeStats, ProgressTracker} from './common';
 
 const ASYNC_TASK_DELAY = 20;
 
-export interface DebugEvent {
-  eventId: GraphDebugEventId;
-  // An associated numeric value. For example, this may represent a duration in
-  // milliseconds for a timing event.
-  eventValue?: number;
+interface DebugTimingEvent {
+  eventId: GraphDebugTimingEventId;
+  // An associated duration in milliseconds for a timing event.
+  eventValue: number;
 }
 
+interface DebugActionEvent {
+  eventId: GraphDebugActionEventId;
```

Comment:
Instead of `isDebugTimingEventId`, I think it makes more sense to make these discriminate-able by typing differently. 

```ts
interface DebugTimingEvent {
  timingId: GraphDebugTimingEventId;
  eventValue: number;
}

interface DebugActionEvent {
  actionId: GraphDebugActionEventId;
  eventLabel?: string;
}

export function notifyDebugEvent(debugEvent: DebugEvent) {
  if ('timingId' in debugEvent) {...}
  else {...}
  ...
}
```

Would that work?